Cargo.toml

```diff
@@ -42,6 +42,52 @@ default = []
 ## [standard compliant]: https://www.w3.org/TR/xml11/#charencoding
 encoding = ["encoding_rs"]
 
+## This feature enables support for deserializing lists where tags are overlapped
+## with tags that do not correspond to the list.
```

Comment:
Just curious about the use case, is this basically for situations where you're parsing messy handwritten XML / HTML? Generally I would think machine-generated XML wouldn't have this problem.

It might be worth providing that as an example of a situation where you might want to enable it if that is the case.

Reply:
Actually, I cannot remember that would I worked with such XMLs, but all parsers, that I know, very tolerant to overlapped lists (even in `xs:sequence`, although XSD requires exact order of tags. I never seen using `xs:all` that allows arbitrary order). So I thought there should be an option to allow this behavior

Reply:
@dralley I'm trying to parse KML file which can have elements of types Folder, Document and Point interleaved in a single list. I'd like to ideally get them in the same order, so I tried to create an enum covering all three cases, but it didn't work. So this at least gives me an option to parse it as three lists without the preservation of order in-between the items.

Reply:
@Mingun it works, but it isn't really what's needed here. What is the desired way to handle things:

input: `<TagOne></TagOne><TagTwo></TagTwo><TagOne></TagOne>`

parsed as:

```
struct TagOne {};
struct TagTwo {};
enum Tag {
  TagOne(TagOne),
  TagTwo(TagTwo)
}

-- result should be
vec![Tag::TagOne(TagOne {}), Tag::TagTwo(TagTwo {}), Tag::TagOne(TagOne {})]
```

I didn't see a way to do something like this via the current lib ^

Do I need to create a separate issue to make something like this possible?

Reply:
Yes, this is an error, I filled #500 for it